src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -414,11 +428,64 @@ private static bool TryParseValue(ref Utf8JsonReader reader, [NotNullWhen(true)]
                         break;
                     }
 
-                    // Single-token values
-                    case JsonTokenType.Number:
+                    case JsonTokenType.Null:
+                        if (useArrayPools)
+                        {
+                            if (reader.HasValueSequence)
+                            {
+                                valueSequence = reader.ValueSequence;
+                            }
+                            else
+                            {
+                                valueSpan = reader.ValueSpan;
+                            }
+
+                            break;
+                        }
+
+                        s_nullLiteral ??= CreateForLiteral(JsonConstants.NullValue.ToArray(), reader.TokenType);
+                        document = s_nullLiteral;
+                        return true;
+
                     case JsonTokenType.True:
+                        if (useArrayPools)
+                        {
+                            if (reader.HasValueSequence)
+                            {
+                                valueSequence = reader.ValueSequence;
+                            }
+                            else
+                            {
+                                valueSpan = reader.ValueSpan;
+                            }
+
+                            break;
+                        }
+
+                        s_trueLiteral ??= CreateForLiteral(JsonConstants.TrueValue.ToArray(), reader.TokenType);
+                        document = s_trueLiteral;
+                        return true;
+
                     case JsonTokenType.False:
-                    case JsonTokenType.Null:
+                        if (useArrayPools)
+                        {
```

Comment:
Rather than expanding this code from one block to three, consider leaving it as one block that just does

```C#
                        if (useArrayPools)
                        {
                            if (reader.HasValueSequence)
                            {
                                valueSequence = reader.ValueSequence;
                            }
                            else
                            {
                                valueSpan = reader.ValueSpan;
                            }

                            break;
                        }

                        document = CreateForLiteral(reader.TokenType);
                        return true;
```

Then, in addition to keeping the code more compact, there's no chance that `CreateForLiteral` is passed a mismatched token type and value pair, and it can do the switch for which field to update internally.